apps/shell/utils/helpers.js

```diff
@@ -1,10 +1,105 @@
-const minimatch = require('minimatch')
+const minimatch = require('minimatch');
+const glob      = require('glob');
+const path      = require('path');
+const yaml      = require('js-yaml');
+const fs        = require('fs');
 
+/**
+ * Generates a set from ood_sshhost_allowlist if exists
+ * 
+ * @param {string} ood_sshhost_allowlist 
+ */
+function generate_host_allowlist(ood_sshhost_allowlist){
+  let host_allowlist = new Set;
+  if (ood_sshhost_allowlist){
+    host_allowlist = new Set(ood_sshhost_allowlist.split(':'));
+  }
+  return host_allowlist;
+}
+
+/**
+ * Returns an array of hashes with information about each cluster
+ * 
+ * @param {string} clusters_d_path process.env.OOD_CLUSTERS if written
+ */
+function generate_cluster_sshhosts(clusters_d_path){
+  let arr = [];
+  glob.sync(path.join((clusters_d_path || '/etc/ood/config/clusters.d'), '*.y*ml'))
+  .map(yml => yaml.safeLoad(fs.readFileSync(yml)))
+  .filter(config => (config.v2 && config.v2.login && config.v2.login.host) && ! (config.v2 && config.v2.metadata && config.v2.metadata.hidden))
+  .forEach((config) => {
+    cluster_info = {}
+    cluster_info['host'] = config.v2.login.host;
+    cluster_info['default'] = config.v2.login.default;
+    arr.push(cluster_info);
```

Comment:
The forEach here could be a map instead:

```
.map(config => { host: config.v2.login.host, default: config.v2.login.default});
```

Then you can get rid of the `let arr = [];` and the `return arr;` and put a return in front of the glob.sync(...